zipkin-core/src/main/scala/org/http4s/zipkin/core/ZipkinInfo.scala

```diff
@@ -0,0 +1,40 @@
+package org.http4s.zipkin.core
+
+import java.time.{Duration, Instant}
+
+import org.http4s.headers.{`X-B3-ParentSpanId`, `X-B3-SpanId`, `X-B3-TraceId`}
+
+final case class Endpoint(
+  ipv4: String,
+  ipv6: Option[String], // TODO: Currently not used
+  port: Int,
+  serviceName: String
+)
+
+final case class Annotation(
+  timestamp: Instant,
+  value: AnnotationType,
+  endpoint: Endpoint
+)
+
+final case class BinaryAnnotation(/* TODO: stuff here */)
+
+sealed trait AnnotationType extends Product with Serializable
+object AnnotationType {
+  case object ClientSend extends AnnotationType
+  case object ServerReceive extends AnnotationType
+  case object ServerSend extends AnnotationType
+  case object ClientReceive extends AnnotationType
+}
+
+final case class ZipkinInfo(
```

Comment:
I'd plan to separate X-B3 eventually from this, particularly as if you end up integrating with cassandra there is a binary form. We're actually making a library for propagated trace context (which you can look at, but obviously doesn't match style) https://github.com/openzipkin/b3-propagation/pull/2